### PR TITLE
Add support for pkg-config

### PR DESCRIPTION
**Pull request by michalbiesek**
### Description
ref #204 

### Types of changes

- [ ] Bugfix (non-breaking change which fixes issue linked in Description above)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)
- [x] Other

### Checklist

- [x] Code compiles without errors
- [x] All tests pass locally with my changes (see TESTING section in CONTRIBUTING file)
- [ ] Created tests which will fail without the change (if possible)
- [x] Extended the README/documentation (if necessary)
- [ ] All newly added files have proprietary license (if necessary)
- [x] All newly added files are referenced in MANIFEST files (if necessary)

## Further comments

---
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/205)

### CODE REVIEW

File: memkind.spec.mk [hunk 1]
```diff
@@ -46,7 +46,7 @@ License: BSD-2-Clause
 Group: System Environment/Libraries
 URL: http://memkind.github.io/memkind
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
-BuildRequires: automake libtool gcc-c++ unzip
+BuildRequires: automake libtool gcc-c++ pkgconfig unzip
```
[the remark this answers is not in the document]
Comment (michalbiesek) [reply memkind.spec.mk]: Removed

File: README [hunk 2]
```diff
@@ -77,6 +77,7 @@ You will need to install required packages on the build system:
 * **libnuma-devel**
 * **libtool**
 * **numactl-devel**
+* **pkg-config**
```
[the remark this answers is not in the document]
Comment (michalbiesek) [reply README]: Removed

File: memkind.pc.in [hunk 3]
```diff
@@ -0,0 +1,12 @@
+version=@VERSION@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: memkind
```
[the remark this answers is not in the document]
Comment (michalbiesek) [reply memkind.pc.in]: I am not an expert for pkg-config, but I am not sure about this libmemkind change:
- see for example: https://github.com/jemalloc/jemalloc/blob/dev/jemalloc.pc.in (here is jemalloc - not libjemalloc)
- and https://repology.org/project/memkind/versions it refers here to memkind not libmemkind

[the remark this answers is not in the document]
Comment (marcinslusarz) [reply memkind.pc.in]: Sure, you can leave it as is, I don't have a strong opinion here.

FTR, the name of the distribution package and the pkg-config name does not have to match. (Different distributions may package software differently, but they can't change the name of the pc file - it must be the same everywhere).

https://people.freedesktop.org/~dbn/pkg-config-guide.html#writing

**Pull request merged**